### PR TITLE
feat[dace]: Buildflags to the `ITIR -> SDFG` translation

### PR DESCRIPTION
**Pull request by philip-paul-mueller**
It makes it possible to pass build flags to the `run_dace_iterator` arguments.

### CODE REVIEW

**Review by edopao — APPROVED** [review]

**Comment by edopao** [comment]
cscs-ci run

File: src/gt4py/next/program_processors/runners/dace_iterator/__init__.py [hunk 1]
```diff
@@ -311,6 +315,7 @@ def _run_dace_cpu(program: itir.FencilDefinition, *args, **kwargs) -> None:
         **kwargs,
         build_cache=_build_cache_cpu,
         build_type=_build_type,
+        build_args=_cpu_args,
```
Comment (edopao) [inline src/gt4py/next/program_processors/runners/dace_iterator/__init__.py:318]: I propose to move the definition of `_cpu_args` inside this function (`_run_dace_cpu`).

I also propose this modification to its definition:
```
compiler_args = dace.config.Config.get("compiler", "cpu", "args")
# disable finite-math-only in order to support isfinite/isinf/isnan builtins
if "-ffast-math" in compiler_args:
    compiler_args += " -fno-finite-math-only"
if "-ffinite-math-only" in compiler_args:
    compiler_args.replace("-ffinite-math-only", "")
```

Comment (edopao) [inline src/gt4py/next/program_processors/runners/dace_iterator/__init__.py:318]: I propose to move the definition of `_cpu_args` inside this function (`_run_dace_cpu`).

I also propose this modification to its definition:
```
compiler_args = dace.config.Config.get("compiler", "cpu", "args")
# disable finite-math-only in order to support isfinite/isinf/isnan builtins
if "-ffast-math" in compiler_args:
    compiler_args += " -fno-finite-math-only"
if "-ffinite-math-only" in compiler_args:
    compiler_args.replace("-ffinite-math-only", "")
```

**Review by edopao — CHANGES_REQUESTED** [review]

**Comment by philip-paul-mueller** [comment]
cscs-ci run

**Review by edopao — APPROVED** [review]

**Pull request merged**